tests/test_detectors.py

```diff
@@ -47,6 +47,38 @@ def test_detector():
         assert th_detector.detect(i) == BasicState([i if i < 2 else 1])
 
 
+def test_interleaved_ppnr():
+    s = [BasicState([i]) for i in range(7)]
+    detector_2wires = Detector.ppnr(n_wires=2)  # The detector contains 2 independent detecting-able systems
+    res = detector_2wires.detect(2)  # 2 photons simultaneously hit the PPNR detector
+    assert s[0] not in res  # At least one photon gets detected => |0> does not apper
+    assert s[1] in res and res[s[1]] == pytest.approx(0.5)
+    assert s[2] in res and res[s[2]] == pytest.approx(0.5)
+    assert len(res) == 2  # Can't detect 3 simultaneous photons with only 2 wires
+
+    res = detector_2wires.detect(3)  # Now with 3 incoming photons
+    assert s[0] not in res
+    assert s[1] in res and res[s[1]] == pytest.approx(0.25)
+    assert s[2] in res and res[s[2]] == pytest.approx(0.75)
+    assert len(res) == 2
+
+    detector_5wires = Detector.ppnr(n_wires=5)  # Adding wires increases the probability of catching all photons
+    res = detector_5wires.detect(3)
+    assert s[0] not in res
+    assert s[1] in res and res[s[1]] == pytest.approx(0.04)
```

Comment:
Was this computed by hand or an independent simulation ?

Reply:
It was by hand, on a paper :/